### PR TITLE
ANC register fixes

### PR DESCRIPTION
**Pull request by ningosi**
@musoke I tried fixing the rows in the template to be like the paper register. Find time and review those. Thanks

### CODE REVIEW

**Comment by ningosi** [comment]
I need this merged so that I can clear conflicts in other reports! @ssmusoke

File: api/src/main/java/org/openmrs/module/ugandaemrreports/data/converter/IYCFataConverter.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+import org.openmrs.Concept;
+import org.openmrs.Obs;
+import org.openmrs.module.reporting.data.converter.DataConverter;
+import org.openmrs.module.ugandaemrreports.reporting.metadata.Dictionary;
+
+/**
+ * Created by Nicholas Ingosi on 5/29/17.
+ */
+public class IYCFataConverter implements DataConverter{
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/data/converter/IYCFataConverter.java]: Rename to IYCFDataConverter

File: api/src/main/resources/org/openmrs/module/ugandaemrreports/reports/.~lock.ANCRegister.xls# [hunk 2]
```diff
@@ -0,0 +1 @@
+,codehub,codehub,30.05.2017 10:41,file:///home/codehub/.config/libreoffice/4;
```
Comment (ssmusoke) [inline api/src/main/resources/org/openmrs/module/ugandaemrreports/reports/.~lock.ANCRegister.xls#]: Remove this file

**Comment by ningosi** [comment]
that sorted @ssmusoke

File: api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetupANCRegister.java [hunk 3]
```diff
@@ -198,43 +193,56 @@ private DataSetDefinition dataSetDefinition() {
         //start constructing of the dataset
         PersonAttributeType attribute = Context.getPersonService().getPersonAttributeTypeByUuid("14d4f066-15f5-102d-96e4-000c29c2a5d7");
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetupANCRegister.java]: @ningosi Rename this attribute variable to reflect the actual attribute

Comment (carapai) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetupANCRegister.java]: @ningosi 
General observations
All ages are supposed be calculated based on the period the report is run and not based on the current date. See AgeLimitCalculation.

Comment (ningosi) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetupANCRegister.java]: @carapai take time and study [this](https://github.com/METS-Programme/openmrs-module-ugandaemr-reports/blob/master/api/src/main/java/org/openmrs/module/ugandaemrreports/definition/data/definition/CalculationDataDefinition.java). All calculations are evaluated and paramaters passed from the evaluation context. Here we base on **onDate** which is mapped to endDate. Note how that parameter is evaluated [here](https://github.com/METS-Programme/openmrs-module-ugandaemr-reports/blob/master/api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetupANCRegister.java#L209) and passed [here](https://github.com/METS-Programme/openmrs-module-ugandaemr-reports/blob/master/api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetupANCRegister.java#L149-L155)

**Pull request merged**